Remove support for tags, also remove the tags column from the member table

whirm opened the pull request:
Malicious proof & co. is also gone.

Commit:
Remove support for tags, also remove the tags column from the member table.

Commit:
Drop malicious_proof table

Commit:
Further removal of malicious behaviour detection code.

NielsZeilemaker commented on community.py:
```diff
@@ -1956,15 +1943,13 @@ def on_messages(self, messages):
         The messages are processed with the following steps:
 
-         1. Messages created by a member in our blacklist are droped.
+         1. Messages that are old or duplicate, based on their distribution policy, are dropped.
```
lets say replaced or superceded instead of old

NielsZeilemaker commented on dispersydatabase.py:
```diff
@@ -380,6 +374,33 @@ def check_database(self, database_version):
+                logger.debug("upgrade database %d -> %d (done)", database_version, 19)
+
+            if database_version < 19:
                 # there is no version 18 yet...
```
19

NielsZeilemaker commented on member.py:
```diff
@@ -220,50 +213,16 @@ def has_identity(self, community):
-        return True
-
+    # TODO(emilon): decide what to do with this
     @property
```
remove it

NielsZeilemaker commented on tests/test_undo.py:
```diff
@@ -191,99 +191,6 @@ def test_node_undo_other(self):
         self._dispersy.get_community(community.cid).unload_community()
 
     @call_on_dispersy_thread
-    def test_self_malicious_undo(self):
```
keep this test, rename it to test_undo_twice

Commit:
Drop the malicious undo blacklisting from check_undo, just keep one of
the messages, and send the message back to the sender in case we are
keeping a different one.

Commit:
Remove must_store attrs.

Commit:
Remove obsoleted tests and fix the failing ones

whirm commented:
This should be ready to be merged

whirm commented:
retest this please

whirm commented:
@NielsZeilemaker, @LipuFei This should be ready to be merged
